modules/openthread/platform/diag.c

```diff
@@ -68,3 +73,151 @@ void otPlatDiagAlarmCallback(otInstance *aInstance)
 {
 	ARG_UNUSED(aInstance);
 }
+
+/*
+ * To enable gpio diag commands, create `openthread-gpio-ports` property
+ * in `zephyr,user` node.
+ * Property should contain phandles of gpio port nodes, e.g:
+ * zephyr,user {
+ *		openthread-gpio-ports = <&gpio0 &gpio1>;
+ * };
+ *
+ * To enable reading current gpio pin mode, define
+ * `CONFIG_GPIO_GET_DIRECTION` in prj.conf.
+ *
+ * Note: `<gpio>` in `diag gpio` commands is an integer that combines
+ * port and pin into a single, contiguous number space as follows:
+ * gpio = (port * 32) + pin
+ */
+#if DT_NODE_EXISTS(DT_PATH(zephyr_user)) && \
+	DT_NODE_HAS_PROP(DT_PATH(zephyr_user), openthread_gpio_ports)
+
+#if DT_PROP_LEN(DT_PATH(zephyr_user), openthread_gpio_ports) == 1
+static const struct device *const gpio_port_dev[] = {
+	DT_GPIO_PORT_FROM_PROP_IDX(DT_PATH(zephyr_user), openthread_gpio_ports, 0)};
+#else
+static const struct device *const gpio_port_dev[] = {DT_FOREACH_PROP_ELEM_SEP(
```

Comment:
Can you elaborate on why wouldn't this approach work for `DT_PROP_LEN(DT_PATH(zephyr_user), openthread_gpio_ports) == 1`?

Reply:
According to documentation of ` DT_FOREACH_PROP_ELEM_SEP(node_id,prop,fn,sep)` macro: 
`The "prop" argument must refer to a property with type string, array, uint8-array, string-array, phandles, or phandle-array. It is an error to use this macro with properties of other types.`
If there is only one port in devicetree e.g.:
```
zephyr,user {
		openthread-gpio-ports = <&gpio0>;
	};
```
`prop` is type of phandle (singular), hence `DT_FOREACH_PROP_ELEM_SEP` fails at build

Reply:
It would be nice if we could simply list all devices with a `gpio-controller;` property from `DT_`-macros.

Reply:
Would it be possible to require `openthread-gpio-ports` to be declared as an array?

Reply:
I have discovered that the issue was laying in creating a property in `zephyr,user` node. It's type was changed automatically on build, so if there was only one element, it changed type to `phandle` (singular). Creating a separate node with Devicetree binding has solved this issue.